Dockerfile

```diff
@@ -1,19 +1,41 @@
 # syntax=docker/dockerfile:1
 
+ARG GO_VERSION=1.19.3
+ARG BASE_DEBIAN_DISTRO="bullseye"
+ARG GOLANG_IMAGE="golang:${GO_VERSION}-${BASE_DEBIAN_DISTRO}"
+ARG XX_VERSION=1.1.2
+
+ARG CONTAINERD_VERSION="v1.6.10"
+ARG RUNC_VERSION="v1.1.4"
+ARG TINI_VERSION="v0.19.0"
+ARG ROOTLESSKIT_VERSION="6222b477d4c3ce6eea2bcff0586e43c95d1c0bb7"
+ARG DOCKERCLI_CHANNEL="stable"
+ARG DOCKERCLI_VERSION="v17.06.2-ce"
```

Comment:
Components versions are set in the Dockerfile as they are only intended to be used there. In a follow-up we could just remove https://github.com/moby/moby/tree/master/hack/dockerfile/install but I keep it in the meantime as `Dockerfile.simple` and `Dockerfile.e2e` still depends on it. Let me know if you think we should keep these installers.

Reply:
Looks like docker-ce-packaging as well: https://github.com/search?q=repo%3Adocker%2Fdocker-ce-packaging%20dockerfile%2Finstall&type=code.

Reply:
Do we re-use these args, or are they only used in the single stage? Wondering if we should just set the value within the stage where they're used.

Reply:
We _could_ consider to remove the default value from the install files, and instead produce an error if no version was set. We probably should move (or copy) the comment from those files into the dockerfile, e.g.;

https://github.com/moby/moby/blob/85b80ce3bb68fe3fe78d70fcf1944ff598649e7f/hack/dockerfile/install/containerd.installer#L4-L18

Reply:
> Do we re-use these args, or are they only used in the single stage?

Not it's just ti have them in the same place but looking at what you said in https://github.com/moby/moby/pull/44513#discussion_r1031523876, I will move them in their respective stage.